optimum/onnxruntime/modeling_decoder.py

```diff
@@ -459,21 +459,24 @@ def _from_pretrained(
             _file_name = model_files[0].name
             subfolder = model_files[0].parent
 
-            defaut_file_name = file_name or "model.onnx"
-            for file in model_files:
-                if file.name == defaut_file_name:
-                    _file_name = file.name
-                    subfolder = file.parent
-                    break
-
-            file_name = _file_name
-
-            if len(model_files) > 1:
-                logger.warning(
-                    f"Too many ONNX model files were found in {' ,'.join(map(str, model_files))}. "
-                    "specify which one to load by using the `file_name` and/or the `subfolder` arguments. "
-                    f"Loading the file {file_name} in the subfolder {subfolder}."
-                )
+            if file_name is not None:
+                _file_name = file_name
```

Comment:
this part of the code is used to find the subfolder associated to the `file_name`, with your modifications loading will fail if there is multiple subfolder with onnx models and if the later is not correctly specified, for example with this modification this loading will fail :

```python
model = ORTModelForCausalLM.from_pretrained(
    "optimum-internal-testing/tiny-random-llama",
    revision="merged-onnx",
    file_name="decoder_model_with_past.onnx",
)
```

Reply:
I don't think this code should be modified, please let me know if there is a model that is being currently being incorrectly loaded